src/engine/ruleset/decoders/f5/f5-afm.yml

```diff
@@ -0,0 +1,85 @@
+    - event.original: <~>F5@<~tmp> <~kv1> date_time=<@timestamp/%b %d %Y %H:%M:%S> <~kv2>
+
+normalize:
+  - map:
+    - ~kv : +s_concat/$~kv1/ /$~kv2
+    - ~kvMap : +parse_kv/$~kv/=/ /'/'
+    - destination.geo.country_name : $~kvMap.dst_geo
+    - destination.nat.ip : $~kvMap.translated_dest_ip
+    - destination.nat.port : $~kvMap.translated_dest_port
+    - destination.port : $~kvMap.dest_port
+    - event.action : $~kvMap.action
+    - event.code : $~kvMap.errdefs_msgno
+    - event.dataset : f5.bigipafm
+    - event.module : f5
+    - event.provider: +ef_delete
+    - fileset.name : bigipafm
+    - host.ip : $~kvMap.bigip_mgmt_ip
+    - host.name : $~kvMap.hostname
+    - input.type : log
+    - log.level : $~kvMap.severity
+    - network.protocol : $~kvMap.ip_protocol
+    - observer.product : $~kvMap.device_product
+    - observer.type : Firewall
+    - observer.vendor : F5
+    - observer.version : $~kvMap.device_version
+    - related.hosts : +a_append/$~kvMap.hostname
+    - related.ip : +a_append/$~kvMap.bigip_mgmt_ip
+    - related.ip : +a_append/$~kvMap.source_ip
+    - related.ip : +a_append/$~kvMap.translated_dest_ip
+    - related.ip : +a_append/$~kvMap.translated_source_ip
+    - related.user : +a_append/$~kvMap.source_user
+    - rsa.internal.messageid: BIGIP_AFM
+    - rsa.investigations.ec_activity: "Disable"
+    - rsa.investigations.ec_subject: "NetworkComm"
+    - rsa.investigations.ec_theme: "Communication"
+    - rsa.misc.action : +a_append/$~kvMap.action
+    - rsa.misc.context : $~kvMap.context_name
+    - rsa.misc.event_type : $~kvMap.errdefs_msg_name
+    - rsa.misc.obj_name: $~tmp
+    - rsa.misc.policy_name : $~kvMap.acl_policy_name
+    - rsa.misc.rule_name : $~kvMap.acl_rule_name
+    - rsa.misc.severity : $~kvMap.severity
+    - rsa.misc.version : $~kvMap.device_version
+    - rsa.network.alias_host: +a_append/$~kvMap.hostname
+    - rsa.network.vlan : $~kvMap.vlan
+    - rsa.time.event_time_str : $@timestamp
+    - rule.name: $~kvMap.acl_rule_name
+    - service.type : f5
+    - source.geo.country_name : $~kvMap.src_geo
+    - source.ip : $~kvMap.source_ip
+    - source.nat.ip : $~kvMap.translated_source_ip
+    - source.nat.port : $~kvMap.translated_source_port
+    - source.port : $~kvMap.source_port
+    - tags : +a_append/$event.dataset
+    - tags : +a_append/forwarded
+    - user.name : $~kvMap.source_user
+    - ~kv1: +ef_delete
```

Comment:
Improve-desing: you could use only one temporary variable and avoid deleting too many variables.